src/python/pants/engine/build_files.py

```diff
@@ -238,13 +241,22 @@ def filter_build_dirs(address_mapper, build_files):
   return BuildDirs(tuple(Dir(d) for d in dirnames if d not in ignored_dirnames))
 
 
+def _pattern_yielder(pattern):
```

Comment:
Seems like overkill for one use-site. Would just be `(pattern, join(**, pattern))`. Also, it's not generic... it's specific to `descendant_addresses_to_globs`.

Also, you might check whether both are necessary... IIRC, @JieGhost changed `**` to mean "zero or more directories", as git defines it. So just `join(**, pattern)` might be sufficient now.